doc/index.md

```diff
@@ -1032,33 +1053,94 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
 
 ## http.server
 
-This interface is **unstable**.
+The http.server object is used to listen on sockets and respond to all types of HTTP requests. Each client request is threaded and independent of the other client requests. An `http.server` is instantiated by passing a number of options including an `onstream` response handler that processes the HTTP requests and provides a stream mechanism for response. `onstream` can also be used for testing and upgrading a request, with HTTP 1.1 to WebSockets being the notible example. The server can be made non-blocking when used in conjunction with cqueues. 
+
+### `server.listen(options)` <!-- --> {#http.server.listen}
+
+Create a new instance of an HTTP Server by passing in table `options`:
+
+  - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
+  - `.cq` (optional): A cqueues controller to use
+  - `.ctx`: an `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: protocol family
+  - `.host`: address to bind to (required if not `.path`)
+  - `.port`: port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
+  - `.path`: path to UNIX socket (required if not `.host`)
+  - `.v6only`: allow ipv6 only (no ipv4-mapped-ipv6)
+  - `.mode`: fchmod or chmod socket after creating UNIX domain socket
+  - `.mask`: set and restore umask when binding UNIX domain socket
+  - `.max_concurrent`: Maximum number of connections to allow live at a time (default: infinity)
+  - `.onerror`: function that will be called when an error occurs (default: do nothing)
+  - `.onstream`: callback function for handling a new client request
```

Comment:
Need to mention arguments received by callback and that it is allowed to throw an error.